Parrot support for mounting ext images

trshaffer opened the pull request:
This PR adds ext2fs as an optional dependency so that Parrot can mount ext[234] disk images. This could be expanded to support read/write access, but for the moment I kept things RO. Users specify the path to an image file which Parrot loads and exposes as a directory tree under `/ext_%d`. I might adjust how images are specified on the command line and the paths they get to improve ergonomics. I've done enough testing to make sure basic functionality works (look around directories, read files, exec programs, follow symlinks).

See also #1947

dthain commented:
@trshaffer can you post a quick how-to here for creating and mounting an image?
@nhazekam please give this a quick try and see if it works for you.

trshaffer commented:
Since this feature gives RO access, you need to use a machine with root access to make an image.

    truncate -s 64M /tmp/test.img
    mkfs.ext2 /tmp/test.img
    losetup -f /tmp/test.img

At this point, you'll need to figure out which loop device you got, e.g. using `losetup -a`.

    mount /dev/loopn /mnt

Now you can put some stuff in `/mnt`. To clean up the mount,

    umount /mnt
    losetup -d /dev/loopn

Now Parrot can mount you image.

    parrot_run --ext /tmp/test.img -- bash

The image will appear under `/ext_0`.

trshaffer commented:
@nhazekam I just changed the command line interface somewhat. The commit message for 
636b914 explains it. I updated the little howto above (mounts to `/ext` rather than `/ext_0`).

dthain commented:
Any reason that you can't make it just `--ext image=mountpoint` ?

trshaffer commented:
@dthain the current thing is just slightly simpler to implement. But now that you mention it, mirroring `mount (1)` is more intuitive for users. I'll switch that over.

batrick commented:
Just wanted to say this looks like a really cool feature! I imagine this will be used for sandboxing/quotas?

If you are considering using this with HTCondor to transmit inputs/outputs, you might want to check if you can shrink-to-fit images (grow on starting Parrot and shrink when exiting?) before transmitting over the network.

trshaffer commented:
@batrick this is actually related to some work @nhazekam is doing on delivering CVMFS to HPC sites. Next in line is reading SquashFS images through Parrot.

nhazekam commented on parrot/src/pfs_main.cc:
```diff
@@ -282,6 +288,8 @@ static void show_help( const char *cmd )
 	printf( " %-30s Make flock a no-op.\n", "--no-flock");
 	printf("\n");
+	printf("Filesystem Options:\n");
+	printf( " %-30s Mount an ext[234] disk image read-only at /ext_n.\n", "--ext-image=<image>");
```
I believe this option has also changed. I am mounting it using `--ext /tmp/test.img=/cvmfs`
